src/subcommand/miner/metrics.rs

```diff
@@ -45,21 +45,21 @@ impl Metrics {
         self.started.elapsed()
     }
 
-    pub(crate) fn hash_rate(&self) -> f64 {
+    pub(crate) fn hash_rate(&self) -> HashRate {
         let mut state = self.hash_rate.lock();
         let total = self.total_hashes();
         let delta = total.saturating_sub(state.last_total);
         state.smoothed.record(delta as f64, Instant::now());
         state.last_total = total;
-        state.smoothed.value()
+        HashRate(state.smoothed.value())
     }
```

Comment:
This behavior is fine the way it is, still I see there is a small cosmetic anomaly with this where if hashes accumulate before the first `hash_rate()` call, the entire accumulated count is recorded as if it happened in one period, causing an initial spike. 

I was a bit hesitant writing this comment as you may find this unnecessary and to be a bit too much complexity, but we can initialize `last_total = total` on first call to mitigate:
```suggestion
    pub(crate) fn hash_rate(&self) -> HashRate {
        let mut state = self.hash_rate.lock();
        let total = self.total_hashes();
    
        if state.last_total == 0 && total > 0 {
            // First call: initialize without recording
            state.last_total = total;
            return HashRate(0.0);
        }
    
        let delta = total.saturating_sub(state.last_total);
        state.smoothed.record(delta as f64, Instant::now());
        state.last_total = total;
        HashRate(state.smoothed.value())
    }
```

Reply:
The handling of this being the first invocation should probably be handled downstream. If you mask this here it actually makes it harder to handle if metatron ever gets a frontend page.